packages/iov-lisk/src/liskclient.ts

```diff
@@ -0,0 +1,200 @@
+ * Encodes the current date and time as a nonce
+ */
+export function generateNonce(): Nonce {
+  const now = new ReadonlyDate(ReadonlyDate.now());
+  return Parse.timeToNonce(now);
+}
+
+export class LiskClient implements IovReader {
+  private readonly baseUrl: string;
+
+  constructor(baseUrl: string) {
+    this.baseUrl = baseUrl;
+  }
+
+  public disconnect(): void {
+    // no-op
+  }
+
+  public async chainId(): Promise<ChainId> {
+    const url = this.baseUrl + "/api/node/constants";
+    const result = await axios.get(url);
+    const responseBody = result.data;
+    return responseBody.data.nethash;
+  }
+
+  public async height(): Promise<number> {
+    const url = this.baseUrl + "/api/node/status";
+    const result = await axios.get(url);
+    const responseBody = result.data;
+    return responseBody.data.height;
+  }
+
+  public async postTx(bytes: PostableBytes): Promise<BcpTransactionResponse> {
+    const transactionId = JSON.parse(Encoding.fromUtf8(bytes)).id as string;
+    if (!transactionId.match(/^[0-9]+$/)) {
+      throw new Error("Invalid transaction ID");
+    }
+
+    await axios.post(this.baseUrl + "/api/transactions", bytes, {
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+
+    // Sleep some seconds to ensure transaction will be found.
+    // Sleep duration determined by trial and error. 15 seconds was not enough.
+    await new Promise(resolve => setTimeout(resolve, 20 * 1000));
+
+    const result = await axios.get(this.baseUrl + `/api/transactions?id=${transactionId}`);
+    const responseBody = result.data;
+
+    let height: number | undefined;
+    let transactionResultBytes: Uint8Array | undefined;
+    if (responseBody.meta.count === 1) {
+      const transactionResult = responseBody.data[0];
+      height = transactionResult.height;
+      transactionResultBytes = Encoding.toUtf8(JSON.stringify(transactionResult));
+    }
+
+    return {
+      metadata: {
+        status: true, // commit failures should throw
+        height: height,
+      },
+      data: {
+        message: "",
+        txid: Encoding.toAscii(transactionId) as TxId,
+        result: transactionResultBytes || new Uint8Array([]),
+      },
+    };
+  }
+
+  public getTicker(_: TokenTicker): Promise<BcpQueryEnvelope<BcpTicker>> {
+    throw new Error("Not implemented");
+  }
+
+  public getAllTickers(): Promise<BcpQueryEnvelope<BcpTicker>> {
+    throw new Error("Not implemented");
+  }
+
+  public async getAccount(query: BcpAccountQuery): Promise<BcpQueryEnvelope<BcpAccount>> {
+    if (isAddressQuery(query)) {
+      const address = query.address;
+      const url = this.baseUrl + `/api/accounts?address=${Encoding.fromAscii(address)}`;
+      const result = await axios.get(url);
+      const responseBody = result.data;
+
+      const account: BcpAccount = {
+        address: address,
+        name: undefined,
+        balance: [
+          {
+            sigFigs: 8,
+            tokenName: undefined,
```

Comment:
please: `tokenName: "LSK" as TokenTicker`

Reply:
`tokenName` != `tokenTicker`. tokenTicker is set in the line after that: `...Parse.liskAmount(responseBody.data[0].balance),`

I don't know what tokenName is supposed to be, but it is of type `string | undefined`, not `TokenTicker`

Reply:
Its the expanded full name. So BTC is a ticker, Bitcoin is a token name.

In this case, it should be `Lisk`

Reply:
This differentiation is needed because there are duplicate tickers in this world, so the name is a separate identifier that would be displayable to a user.

Reply:
fixed. And I think we can make the field non-optional. But this will go in the next PR https://github.com/iov-one/iov-core/pull/358/commits/1e82ca7b06747a9b22a6d46dd4f8fb1cd1aa8ebd